components/tabs/tabs.js

```diff
@@ -0,0 +1,388 @@
+
+		};
+		*/
+	}
+
+	render() {
+
+		return html`
+			<div class="d2l-tabs-layout d2l-body-compact">
+				<div class="d2l-tabs-container">
+					<div class="d2l-tabs-scroll-previous-container">
+						<button class="d2l-tabs-scroll-button" title="${this.localize('scroll.previous')}"><d2l-icon icon="tier1:chevron-left"></d2l-icon></button>
+					</div>
+					<span class="d2l-tabs-focus-start" @focus="${this._handleFocusStart}" tabindex="0"></span>
+					${this.arrowKeysContainer(html`
+						<div class="d2l-tabs-container-list"
+							@d2l-tab-selected="${this._handleTabSelected}"
+							role="tablist">
+							${repeat(this._tabInfos, (tabInfo) => tabInfo.id, (tabInfo) => html`
+								<d2l-tab aria-selected="${tabInfo.selected ? 'true' : 'false'}"
+									.controlsPanel="${tabInfo.id}"
+									text="${tabInfo.text}">
+								</d2l-tab>
+							`)}
+						</div>
+					`)}
+					<span class="d2l-tabs-focus-end" @focus="${this._handleFocusEnd}" tabindex="0"></span>
+					<div class="d2l-tabs-scroll-next-container">
+						<button class="d2l-tabs-scroll-button" title="${this.localize('scroll.next')}"><d2l-icon icon="tier1:chevron-right"></d2l-icon></button>
+					</div>
+				</div>
+				<div class="d2l-tabs-container-ext"><slot name="ext"></slot></div>
+			</div>
+			<div class="d2l-panels-container"
+				@d2l-tab-panel-selected="${this._handlePanelSelected}"
+				@d2l-tab-panel-text-changed="${this._handlePanelTextChange}">
+				<slot @slotchange="${this._handlePanelsSlotChange}"></slot>
+			</div>
+		`;
+	}
+
+	_focusSelected() {
+		const selectedTab = this.shadowRoot.querySelector('d2l-tab[aria-selected="true"]');
+		if (selectedTab) {
+			/*
+				update the virtual scroll position so that the browser doesn't force a scroll then...
+			*/
+			selectedTab.focus();
+		}
+	}
+
+	_getComputedBackgroundColor() {
+		let bgColor = null;
+
+		findComposedAncestor(this, (node) => {
+			if (!node || node.nodeType !== Node.ELEMENT_NODE) return false;
+			const nodeColor = getComputedStyle(node, null)['backgroundColor'];
+			if (nodeColor === 'rgba(0, 0, 0, 0)' || nodeColor === 'transparent') return false;
+			bgColor = nodeColor;
+			return true;
+		});
+
+		return bgColor;
+	}
+
+	_getPanel(id) {
+		// use simple selector for slot (Edge)
+		const slot = this.shadowRoot.querySelector('.d2l-panels-container').querySelector('slot');
+		const panels = this._getPanels(slot);
+		for (let i = 0; i < panels.length; i++) {
+			if (panels[i].nodeType === Node.ELEMENT_NODE && panels[i].role === 'tabpanel' && panels[i].id === id) {
+				return panels[i];
+			}
+		}
+	}
+
+	_getPanels(slot) {
+		if (!slot) return;
+		return slot.assignedNodes({flatten: true})
+			.filter((node) => node.nodeType === Node.ELEMENT_NODE && node.role === 'tabpanel');
+	}
+
+	_getTabInfo(id) {
+		for (let i = 0; i < this._tabInfos.length; i++) {
+			if (this._tabInfos[i].id === id) return this._tabInfos[i];
+		}
```

Comment:
Nit, could use:

```
return this._tabInfos.find( (t) => t.id === id );
```

Although if it doesn't find anything, `undefined` is returned not `null`.
```

Reply:
Yeah I was considering that, but elected not to use it since IE11 is in limbo...?

Reply:
Pretty sure it'll get Babelified for IE11.